backend/src/apiserver/archive/log.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package archive
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+)
+
+type LogFormat string
+
+const (
+	LogFormatJSON = LogFormat("json-lines")
+	LogFormatText = LogFormat("text")
+)
+
+type ExtractLogOptions struct {
+	LogFormat  LogFormat
+	Timestamps bool
+}
+
+type LogArchiveInterface interface {
+	GetLogObjectKey(workflow *util.Workflow, nodeId string) (string, error)
+	CopyLogFromArchive(logContent []byte, dst io.Writer, opts ExtractLogOptions) error
+}
+
+// Log Archive
+type RunLogEntry struct {
+	Log       string    `json:"log"`
+	Timestamp time.Time `json:"timestamp,omitempty"`
+}
+
+var k8sLogPrefixExp = regexp.MustCompile(`(?m)^(\d{4}-\d{2}-\d{2}T\S+)\s(.+)$`)
+var crioLogPrefixExp = regexp.MustCompile(`(?m)^(.+)\s(stdout|stderr)\s\w\s(.+)$`)
```

Comment:
I think you mentioned some of these lines of code were "copied" from a library with similar functionality. Can you include its full license file around those lines?

Reply:
I mentioned about fluent-bit parsers and referenced Apache 2.0 licence